feature_engine/sanity_check/similar_columns.py

```diff
@@ -0,0 +1,162 @@
+from typing import Any, List
+import logging
+
+import numpy as np
+import pandas as pd
+from sklearn.base import BaseEstimator, TransformerMixin
+from sklearn.utils.validation import check_is_fitted
+
+from feature_engine.dataframe_checks import (
+    _check_contains_inf,
+    _check_contains_na,
+    _is_dataframe,
+)
+
+
+class SimilarColumns(BaseEstimator, TransformerMixin):
+    """Ensure that similar columns are in test and train dataset.
+
+
+    Parameters
+    ----
+    None
+
+    """
+
+    def __init__(
+        self,
+        impute_with: Any = np.nan,
+        missing_values: str = "ignore",
+        drop_if_more_columns: bool = True,
+        add_if_less_columns: bool = True,
+        verbose: bool = False
+    ):
+
+        if missing_values not in ["raise", "ignore"]:
+            raise ValueError("missing_values takes only values 'raise' or 'ignore'.")
+
+        if not isinstance(drop_if_more_columns, bool):
+            raise ValueError("drop_if_more_columns takes only booleans True and False")
+
+        if not isinstance(add_if_less_columns, bool):
+            raise ValueError("add_if_less_columns takes only booleans True and False")
+
+        if not isinstance(verbose, bool):
+            raise ValueError("verbose takes only booleans True and False")
+
+        self.impute_with = impute_with
+        self.missing_values = missing_values
+        self.drop_if_more_columns = drop_if_more_columns
+        self.add_if_less_columns = add_if_less_columns
+        self.verbose = verbose
+
+    def _check_input(self, X: pd.DataFrame):
+        X = _is_dataframe(X)
+
+        if self.missing_values == "raise":
+            # check if dataset contains na
+            _check_contains_na(X, self.variables_)
+            _check_contains_inf(X, self.variables_)
+
+        return X
+
+    def fit(self, X: pd.DataFrame, y: pd.Series = None):
+        """Fit columns schema
+
+        Parameters
+        ----------
+
+        X: pandas dataframe of shape = [n_samples, n_features]
+            The input dataframe
+
+        y: None
+            y is not needed for this transformer. You can pass y or None.
+
+        """
+        X = self._check_input(X)
+
+        self.variables_ = X.columns
+        return self
+
+    def _raise_errors_if_needed(
+        self,
+        _columns_to_drop: List[str],
+        _columns_to_add: List[str],
+        X: pd.DataFrame
+    ):
+
+        if not self.drop_if_more_columns and len(_columns_to_drop) > 0:
+            raise ValueError(
```

Comment:
I think it should stay an error, sklearn transformer have check that raise errors if their is extra columns in transform df than in fit df.
Yes the transformer doesn't work if the User select False and there is extra columns, but it works if the user select False and the number of columns is equal or inferior (if User select true for  add_if_less_columns) to fit df.

Reply:
@NicoGalli

Reply:
@solegalli @thibaultbl I would suggest to keep it as simple as possible, at least at the bigining. If it is unclear this validation is needed or not, I would suggest not to include it right now and simplify the amount of paraments for the users to implement this Class.  If you guys would like me to dig deeper, just let me know.

Reply:
I agree with @NicoGalli 

Thinking it further, the whole point of this new transformer is to make coming dataframes match the variables of the train set, so that is what the functionality should be.

Allowing the user to not add or not remove some variables, defeats the purpose of the transformer imho.